cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -2252,3 +2264,16 @@ def remove_hacluster():
         push_service_data()
 
     clear_flag('hacluster-configured')
+
+
+def get_dns_provider():
+    dns_provider = hookenv.config('dns-provider').lower()
+    if dns_provider == 'auto':
+        dns_provider = leader_get('auto_dns_provider')
+        if not dns_provider:
```

Comment:
Perhaps a comment here to explain when this condition will be hit, as it wasn't immediately obvious to me.

Reply:
Added a comment. Thanks for the comment.